spec/units.yaml

```diff
@@ -1,238 +1,236 @@
 # This file defines units supported in the VSS standard catalog
+# So far only two domains defined, ides is to have domains for all
+# Criteria: Two units belongs to same domain if it is possible to convert
+# between them, either with multiplier/denominator or by custom
+# (hard-coded) function
+#
+# (However, conversion may result in out of range if target unit/datatype cannot
+# represent the value)
+
+domains:
+  distance:
+    description: Length traveled or length between two objects
+    # Default unit specifies the base for conversion
+    # Other units needs to specify how conversion to default unit can happen
+    default-unit: m
+  speed:
+    description: Change in distance per time unit
+    default-unit: km/h
+  time:
+    description: Specific moment in time
+    default-unit: iso8061
+  duration:
+    description: Time between two events
+    default-unit: s
 units:
   mm:
-    label: millimeter
     description: Distance measured in millimeters
     domain: distance
+    # Multiply this value by 1000 to get default unit (meters)
+    multiplier: 1000
+    # List of allowed datatypes. Numeric is short-cut for all int/uint types
+    allowed-datatypes: ['numeric']
```

Comment:
The idea here is to give a warning or error if someone defines a VSS signal that has the unit `mm` but where the datatype is `string`, `boolean` or some struct type. Array types based on any numeric type should be accepted.